lock-api-objects/src/main/java/com/palantir/lock/watch/IdentifiedVersion.java

```diff
@@ -16,14 +16,28 @@
 
 package com.palantir.lock.watch;
 
+import java.util.Comparator;
 import java.util.UUID;
 
 import org.immutables.value.Value;
 
+import com.palantir.logsafe.Preconditions;
+
 @Value.Immutable
 public interface IdentifiedVersion {
     @Value.Parameter
     UUID id();
     @Value.Parameter
     long version();
+
+    static IdentifiedVersion of(UUID id, long version) {
+        return ImmutableIdentifiedVersion.of(id, version);
+    }
+
+    static Comparator<IdentifiedVersion> comparator() {
+        return (version1, version2) -> {
+            Preconditions.checkArgument(version1.id().equals(version2.id()), "Versions do not have same leader id");
```

Comment:
I don't feel like this comparator should be here: it's a pretty specific assumption/underlying design decision in your LockWatchEventCacheImpl, so should be implemented there.

Reply:
I use it once in `ClientLockWatchEventLogImpl` too, hence my reluctance to remove from that class.

Reply:
And now is used across three classes (since I split it out), so I will leave there.